### PR TITLE
cli: convert to using new sdk client

### PR DESCRIPTION
**Pull request by bmwill**
* 934a4fea06 cli: convert to use new sdk client
* 236df5a013 client: add retry logic
* 550dcf71c7 client: add wait_for_transaction method
* 3fa73d67e6 client: add state ratcheting

### CODE REVIEW

File: sdk/client/src/blocking.rs [hunk 1]
```diff
@@ -1,31 +1,104 @@
...
+            txn.sender(),
+            txn.sequence_number(),
+            txn.expiration_timestamp_secs(),
+            &Transaction::UserTransaction(txn.clone()).hash().to_hex(),
+            timeout,
+            delay,
+        )
+    }
+
+    fn wait_for_transaction_impl(
```
Comment (xli) [inline sdk/client/src/blocking.rs:65]: we probably should public this method too, for the case only (address, seq, hash) tuple is saved other than the whole SignedTransaction

File: sdk/client/src/macros.rs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    ($($item:item)*) => {
+        $(
+            #[cfg(feature = "blocking")]
+            #[cfg_attr(docsrs, doc(cfg(feature = "blocking")))]
+            $item
+        )*
+    }
+}
+
+macro_rules! cfg_async_or_blocking {
```
Comment (xli) [inline sdk/client/src/macros.rs:24]: what is reason we need this macro? only expose structs when non of async / blocking feature is enabled?

**Review by xli — APPROVED** [review]

File: sdk/client/src/macros.rs (hunk 2 above)
Comment (bmwill) [reply sdk/client/src/macros.rs:24]: yeah exactly, there are some things that are shared between the two impls so they need to be enabled when either is enabled.

**Pull request merged**